src/services/projects.js

```diff
@@ -318,4 +321,280 @@ module.exports = class ProjectsHelper {
+							}
+							if (entityType.value !== common.TASKS) {
+								let checkForRegex = utils.checkRegexPattarn(projectEntityData, entityType.validations)
+								if (checkForRegex) {
+									return responses.failureResponse({
+										message: `Special characters not allowed in ${entityType.value}`,
+										statusCode: httpStatusCode.bad_request,
+										responseCode: 'CLIENT_ERROR',
+										error: utils.errorObject(common.BODY, entityType.value),
+									})
+								}
+							}
+						}
+					} else {
+						if (entityType.has_entities) {
+							if (Array.isArray(projectEntityData)) {
+								// If projectEntityData is an array, check if all items are present in allEntities
+								for (const item of projectEntityData) {
+									const entitiesPresent = allEntities.result.find(
+										(entity) => entity.value === item.value
+									)
+									if (!entitiesPresent) {
+										return responses.failureResponse({
+											message: `invalid ${entityType.value} added`,
+											statusCode: httpStatusCode.bad_request,
+											responseCode: 'CLIENT_ERROR',
+											error: utils.errorObject(common.BODY, entityType.value),
+										})
+									}
+								}
+							} else if (typeof projectEntityData === common.OBJECT) {
+								// If projectEntityData is an object, check if the item is present in allEntities
+								const entitiesPresent = allEntities.result.find(
+									(entity) => entity.value === projectEntityData.value
+								)
+								if (!entitiesPresent) {
+									return responses.failureResponse({
+										message: `invalid ${entityType.value} added`,
+										statusCode: httpStatusCode.bad_request,
+										responseCode: 'CLIENT_ERROR',
+										error: utils.errorObject(common.BODY, entityType.value),
+									})
+								}
+							}
+						} else {
+							// If has_entities is false, check if the key in projectData has a non-empty value
+
+							if (entityType.value !== common.TASKS && entityType.value) {
+								let checkForRegex = utils.checkRegexPattarn(projectEntityData, entityType.validations)
+								if (checkForRegex) {
+									return responses.failureResponse({
+										message: `Special characters not allowed in ${entityType.value}`,
+										statusCode: httpStatusCode.bad_request,
+										responseCode: 'CLIENT_ERROR',
+										error: utils.errorObject(common.BODY, entityType.value),
+									})
+								}
+							}
+						}
+					}
+				}
+
+				if (projectData.tasks.length < 1) {
+					return responses.failureResponse({
+						message: 'TASK_NOT_FOUND',
+						statusCode: httpStatusCode.bad_request,
+						responseCode: 'CLIENT_ERROR',
+						error: utils.errorObject(common.BODY, 'tasks'),
+					})
+				}
+				let taskEntityTypes = await entityModelMappingQuery.findEntityTypes(
+					{
+						model: common.TASKS,
+						status: common.STATUS_ACTIVE,
+					},
+					['value', 'validations']
+				)
+				for (let i = 0; i < projectData.tasks.length; i++) {
```

Comment:
@ankit1shahu instead of i use some value like the above comment or use 
 for (const task of projectData.tasks) {
      for (const taskEntityType of taskEntityTypes) {